tools/base/fpgainfo/bmcdata.c

```diff
@@ -111,24 +112,36 @@ fpga_result get_metrics(fpga_token token,
 }
 
 void print_metrics(const fpga_metric_info *metrics_info,
+                    uint64_t num_metrics_info,
                    const fpga_metric *metrics, uint64_t num_metrics)
 {
-    uint64_t i = 0;
-    for (i = 0; i < num_metrics; ++i) {
-        uint64_t idx = metrics[i].metric_num;
-        
-        if (idx < num_metrics) {
-            printf("%-29s : ", metrics_info[idx].metric_name);
-
-            if (metrics_info[i].metric_datatype == FPGA_METRIC_DATATYPE_INT) {
-                printf("%" PRId64 "", metrics[i].value.ivalue);
-            } else if (metrics_info[i].metric_datatype == FPGA_METRIC_DATATYPE_FLOAT ||
-                    metrics_info[i].metric_datatype == FPGA_METRIC_DATATYPE_DOUBLE) {
-                printf("%0.2f", metrics[i].value.dvalue);
-            }
+	uint64_t i = 0;
+	for (i = 0; i < num_metrics; ++i) {
+		uint64_t idx = metrics[i].metric_num;
 
-            printf(" %s\n", metrics_info[idx].metric_units);
-        }
-    }
+		if (metrics[i].isvalid) {
+
+			if (idx < num_metrics) {
+				printf("(%2ld)  %-29s : ", i + 1, metrics_info[idx].metric_name);
+
+				if (metrics_info[idx].metric_datatype == FPGA_METRIC_DATATYPE_INT) {
+					printf("%" PRId64 "", metrics[i].value.ivalue);
+				}
+				else if (metrics_info[idx].metric_datatype == FPGA_METRIC_DATATYPE_FLOAT ||
+					metrics_info[idx].metric_datatype == FPGA_METRIC_DATATYPE_DOUBLE) {
+					printf("%0.2f", metrics[i].value.dvalue);
+				}
+
+				printf(" %s\n", metrics_info[idx].metric_units);
+			}
+		} else {
+		// Failed to read metric value
+			 if (idx < num_metrics_info) {
```

Comment:
Is this block supposed to be in the `else` block for `if (metrics[i].valid)`? 
What about the `else` condition for `if (idx < num_metrics)`?

Reply:
fixed